PublicMapping/districtbuilder
# Allow adding reference layers to project

## Overview

- Add modal to support uploading a new reference layer
- Add `ReferenceLayer` entity to backend
- Add a new endpoint to parse and validate a geojson or csv upload
- Add new endpoint to save a `reference_layer` for a project
- Display all uploaded reference layers in a new pane in the project sidebar


### Checklist


### Demo

## Create form
![image](https://user-images.githubusercontent.com/66973361/128771012-a05d5410-0988-4069-9e58-66e3bab1262d.png)
![image](https://user-images.githubusercontent.com/66973361/128774053-f4e94490-ae49-42a5-9727-8d590a41de3f.png)
![image](https://user-images.githubusercontent.com/66973361/128771058-a6e83379-c8f9-4df2-9924-a4b405825286.png)

## Displayed in sidebar
![image](https://user-images.githubusercontent.com/66973361/128771792-9a5ef647-5f3a-4ab9-8ec9-72c842edbbd9.png)



### Notes

Still need a bit of polish on the validation and csv support here, but figured I'd get a draft PR up here to begin reviewing since there is a lot here to look at.


## Testing Instructions

- `./scripts/update`
- Open a project, and click the + button in the bottom of the project sidebar on the left hand side to open the reference layer modal
- Attempt to upload the two attached geojson files; expect them to be saved to DB and record displayed in project sidebar

Closes #861
